File: day_1/Solution/day1.cs
var fileContents = File.ReadAllLines("Inputs/day1.txt").ToList();
var list1 = new List<int>();
var list2 = new List<int>();
var totalDistance = 0;

foreach (var line in fileContents)
{
    var splitLine = line.Split("   ").ToList();
    list1.Add(int.Parse(splitLine[0]));
    list2.Add(int.Parse(splitLine[1]));
}

list1.Sort();
list2.Sort();

foreach (var (index, item) in list1.Index())
    totalDistance += Math.Abs(item - list2[index]);

Console.WriteLine(totalDistance);

File: day_1/Solution/day1.rb
# frozen_string_literal: true

column1 = []
column2 = []
total_distance = 0

File.open('../sample_data.txt', 'r') do |file|
  file.each_line do |line|
    values = line.split
    column1 << values[0].to_i
    column2 << values[1].to_i
  end
end

column1.sort!
column2.sort!

column1.each_with_index do |value, index|
  total_distance += (value - column2[index]).abs
end

puts "Total distance: #{total_distance}"